openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -132,12 +146,16 @@ def get_context_data(self, **kwargs):
             # avoiding a double-logout.
             if not referrer or (referrer and not uri.startswith(referrer)):
                 logout_uris.append(self._build_logout_url(uri))
+        if referrer == LEARNING_PORTAL and self.auth_backend == SAML_BACKEND:
+            self.is_saml_logout = True
 
         target = self.target
         context.update({
             'target': target,
             'logout_uris': logout_uris,
             'enterprise_target': self._is_enterprise_target(target),
+            'saml_logout_url': '{}?{}'.format(self.saml_logout_url, urlencode({'redirect_uri': settings.LMS_ROOT_URL})),
```

Comment:
This line does not make sense. why we are prefixing IDP's logout url with LMS base url?

Reply:
After successfully logout from IDP ( Keyclock in our case) needs to redirect back to LMS home page. This line added `redirect_uri` query_param with IDP's `logout_url`.